engine/engine.go

```diff
@@ -14,25 +14,31 @@ import (
 	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promauto"
+
 	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/prometheus/promql/parser"
 	"github.com/prometheus/prometheus/storage"
 	v1 "github.com/prometheus/prometheus/web/api/v1"
 
+	"github.com/thanos-community/promql-engine/logicalplan"
 	"github.com/thanos-community/promql-engine/physicalplan"
 	"github.com/thanos-community/promql-engine/physicalplan/model"
 	"github.com/thanos-community/promql-engine/physicalplan/parse"
 )
 
 type engine struct {
-	logger        log.Logger
-	debugWriter   io.Writer
-	lookbackDelta time.Duration
+	logger           log.Logger
+	debugWriter      io.Writer
+	lookbackDelta    time.Duration
+	enableOptimizers bool
 }
 
 type Opts struct {
 	promql.EngineOpts
 
+	// DisableOptimizers disables query optimizations using logicalPlan.DefaultOptimizers.
+	DisableOptimizers bool
```

Comment:
My worry is that this has to be extended to have different optimization modes, possible with https://yourbasic.org/golang/bitmask-flag-set-clear/

I guess we can break compatibility still and change it in future if needed.